Don't pack implicit references as framework references

dsplaisted opened the pull request:
- Add Pack=false metadata to implicit framework references for .NET Framework projects
- Don't add implicit references that are already explicitly referenced

Fixes #1179

Note that this updates to a version of NuGet that has the framework reference support in the pack command.  So that update may need to be coordinated across the SDK and CLI repos.

dsplaisted committed:
Add Pack=false metadata to implicit framework references, don't add implicit references that are already explicitly referenced

Fixes #1179

dsplaisted committed:
Add explicit visibility modifier to a few methods

rohit21agrawal commented on build/DependencyVersions.props:
```diff
@@ -10,7 +10,7 @@
     <DependencyModelVersion>2.0.0-preview1-001960</DependencyModelVersion>
     <PlatformAbstractionsVersion>2.0.0-preview1-001960</PlatformAbstractionsVersion>
-    <NuGetVersion>4.3.0-preview1-2500</NuGetVersion>
+    <NuGetVersion>4.3.0-preview1-4045</NuGetVersion>
```
not sure if we want to update this just yet.

CC: @rrelyea for confirmation.

dsplaisted replied:
Yeah, I called that out.  It's needed for the test coverage, so we could split it out if needed.

dsplaisted commented:
@MattGertz for approval

**Customer scenario**

Create a NuGet package targeting .NET Framework

**Bugs this fixes:** 

#1179

**Workarounds, if any**

Set `DisableImplicitFrameworkReferences` to true and explicitly reference the framework assemblies you want.

**Risk**

Low - This adds extra MSBuild metadata to a few Reference items

**Performance impact**

Low - This adds extra MSBuild metadata to a few Reference items

**Is this a regression from a previous update?**

Sort of.  NuGet/NuGet.Client#1361 made it so that NuGet Pack could add framework assembly references to a package.  Since we implicitly add references to .NET Framework assemblies, they would now be included in all packages that target .NET Framework.

**Root cause analysis:**

NuGet/NuGet.Client#1361 made it so that NuGet Pack could add framework assembly references to a package.  Since we implicitly add references to .NET Framework assemblies, they would now be included in all packages that target .NET Framework.

**How was the bug found?**

Discussion related to NuGet/NuGet.Client#1361

dsplaisted committed:
Disable tests targeting .NET Framework on non-Windows

# Conflicts:
#	test/Microsoft.NET.Pack.Tests/GivenThatWeWantToPackANetFrameworkLibrary.cs
